### PR TITLE
Cluedo - Mystery Randomizer by Ylva

### PR DESCRIPTION
**Pull request by ylvan75**
I'm aware of thing if I have  two or more attributes I should add linebreaks to increase readability.

I use Prettier (with default settings) and when I do save it change it back to be on one single line.

For next weeks project I will look into that and see what I need to change in Prettier to be able to add linebreaks.

Happy to get advice what I have to do.

### CODE REVIEW

File: code/clue.js [hunk 1]
```diff
@@ -1,59 +1,325 @@
...
+  id: "pistol",
+  weight: 7,
+};
+
+//Console log information from some of the suspects and weapons objects
+console.log(mrGreen.firstName);
+console.log(professorPlum.lastName);
+console.log(missScarlet.color);
+console.log(rope.name);
+console.log(knife.weight);
```
Comment (sarahejhej) [inline code/clue.js]: Remember to delete all `console.log` before pull request!

File: code/clue.js [hunk 2]
```diff
@@ -1,59 +1,325 @@
...
 
 // THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
 
 // NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
```
Comment (sarahejhej) [inline code/clue.js]: I think the code would be cleaner if instructions like these were deleted. Maybe adding own comment headers to make it easier to navigate in the code? Like variables and functions etc

File: code/clue.js [hunk 3]
```diff
@@ -1,59 +1,325 @@
...
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
```
Comment (sarahejhej) [inline code/clue.js]: Some rooms occur more than once. Is that to make them more likely to be picked?

File: code/clue.js [hunk 4]
```diff
@@ -1,59 +1,325 @@
...
+const toggleAnimationDeckCard = (id) => {
+  document.getElementById(id).classList.toggle("deck-heartbeat");
+};
 
 // This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+const pickKiller = (event) => {
+  event.preventDefault();
```
Comment (sarahejhej) [inline code/clue.js]: This was new to me, `event.preventDefault()`, have to look into it!

File: code/clue.js [hunk 5]
```diff
@@ -1,59 +1,325 @@
...
+      document.getElementById(
+        "killerOccupation"
+      ).innerHTML = `Occupation: ${mystery.killer.occupation}`;
+      document.getElementById("killerImage").src = `${mystery.killer.image}`;
+      document.getElementById(
+        "killerDescription"
+      ).innerHTML = `${mystery.killer.description}`;
+      document.getElementById(
+        "killerFavouriteWeapon"
+      ).innerHTML = `Favourite Weapon: ${mystery.killer.favouriteWeapon}`;
```
Comment (sarahejhej) [inline code/clue.js:218]: I get the same formatting, it makes it so hard to read! If you find out how to change it please let me know

File: code/clue.js [hunk 6]
```diff
@@ -1,59 +1,325 @@
...
-}
+    toggleAnimationDeckCard("killerCard");
+
+    setTimeout(() => {
+      shuffleFavouriteWeapon();
+
+      // This will randomly select a killer from the suspects. And add that to the mystery object.
+      mystery.killer = randomSelector(suspects);
+
+      // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
```
Comment (sarahejhej) [inline code/clue.js]: I think you can delete a few blank lines here, I find it difficult to follow the code with the blank lines inside of a function

File: code/clue.js [hunk 7]
```diff
@@ -1,59 +1,325 @@
...
+      ).innerHTML = `Occupation: ${mystery.killer.occupation}`;
+      document.getElementById("killerImage").src = `${mystery.killer.image}`;
+      document.getElementById(
+        "killerDescription"
+      ).innerHTML = `${mystery.killer.description}`;
+      document.getElementById(
+        "killerFavouriteWeapon"
+      ).innerHTML = `Favourite Weapon: ${mystery.killer.favouriteWeapon}`;
+    }, 2000);
+    killerPicked = true;
```
Comment (sarahejhej) [inline code/clue.js:220]: This I recognise :)

File: code/clue.js [hunk 8]
```diff
@@ -1,59 +1,325 @@
...
+    // This will remove the reveal the crime message if any
+    cleanRevealMysteryMessage();
+
+    toggleAnimationDeckCard("roomCard");
+
+    // This will randomly select a room from the rooms. And add that to the mystery object.
+    setTimeout(() => {
+      mystery.room = randomSelector(rooms);
+
+      // This will change the card to the one connected to the chosen room and show which room it is.
```
Comment (sarahejhej) [inline code/clue.js]: Nice commenting! Makes it so much easier to follow the code!

File: code/clue.js [hunk 9]
```diff
@@ -1,59 +1,325 @@
...
+};
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+const revealMystery = (event) => {
+  event.preventDefault();
+  if (
+    mystery.killer === null ||
+    mystery.weapon === null ||
+    mystery.room === null
```
Comment (sarahejhej) [inline code/clue.js:260]: Always good to see solutions that differ from your own, great way to learn!

File: code/clue.js [hunk 10]
```diff
@@ -1,59 +1,325 @@
...
+    document.getElementById(
+      "mystery"
+    ).innerHTML = `The murder was commited by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} room with a ${mystery.weapon.name}!`;
+  }
+};
+
+document.getElementById("killerCard").addEventListener("click", pickKiller);
+document.getElementById("weaponCard").addEventListener("click", pickWeapon);
+document.getElementById("roomCard").addEventListener("click", pickRoom);
+document.getElementById("revealCard").addEventListener("click", revealMystery);
```
Comment (sarahejhej) [inline code/clue.js:274]: This looks super clean! I struggled to understand the syntax of `.addEventListener`

File: code/index.html [hunk 11]
```diff
@@ -4,33 +4,39 @@
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <meta name="author" content="Ylva Nilsson" />
```
Comment (sarahejhej) [inline code/index.html:7]: Never seen this, cool!

File: code/index.html [hunk 12]
```diff
@@ -4,33 +4,39 @@
...
     <link rel="stylesheet" href="style.css" />
-    <title>Technigo Project Cluedo</title>
+    <link rel="icon" type="image/png" sizes="16x16" href="./assets/loupe.png" />
```
Comment (sarahejhej) [inline code/index.html:9]: I like the icon!

File: code/style.css [hunk 13]
```diff
@@ -1,58 +1,112 @@
...
   background-color: #fff;
 }
 
+h1 {
+  text-align: center;
+}
+
 .cards {
   margin: 50px auto;
   width: 90%;
```
Comment (sarahejhej) [inline code/style.css:39]: Great with mobile first!

File: code/style.css [hunk 14]
```diff
@@ -1,58 +1,112 @@
...
   border: none;
   font-size: 20px;
   border-radius: 8px;
   display: block;
-  margin: 0 auto;
+  margin: 20px auto;
+  box-shadow: 6px 6px 3px 0px rgba(0, 0, 0, 0.3);
+}
+
+@keyframes heartbeat {
```
Comment (sarahejhej) [inline code/style.css:82]: I like the animation!

File: code/style.css [hunk 15]
```diff
@@ -1,58 +1,112 @@
...
+  60% {
+    transform: scale(1.1);
+  }
+  100% {
+    transform: scale(1);
+  }
+}
+
+/***** MEDIA QUERIES - TABLET and DESKTOP ****/
+@media (min-width: 481px) {
```
Comment (sarahejhej) [inline code/style.css:101]: This is so clean - having to add just a few lines to make the site responsive. Yay!

Comment (sarahejhej) [inline code/style.css:101]: The first breakpoint is usually at 768 px

File: code/index.html [hunk 16]
```diff
@@ -4,33 +4,39 @@
...
-        <p id="killerAge"></p>
-        <p id="killerOccupation"></p>
-        <img id="killerImage" src="" height="100" />
-        <p id="killerDescription"></p>
+      <article class="deck killer-deck" id="killerCard">
+        <p class="deck-title">KILLER</p>
+        <p class="deck-content" id="killerName"></p>
+        <p class="deck-content" id="killerAge"></p>
+        <p class="deck-content" id="killerOccupation"></p>
+        <img id="killerImage" src="" height="200" />
```
Comment (sarahejhej) [inline code/index.html:23]: I think the height attribute could be moved into the style.css file

**Review by sarahejhej — CHANGES_REQUESTED** [review]
Well done on this project! Your README.md was very informative and explanatory, and your comment about the formatting in the pull request made it easier to look through the code. 
I know that you feel like you struggle with the design, but I think both the design and the layout are very clean! And I like the colour scheme. The page is also responsive, which is great. It wasn't part of the assignment, but you made the extra effort to do so. And also, it's mobile first 👍

When testing your site I noticed that you made sure that you could't reveal the mystery before all cards were picked. I think that you easily could have added an if-statement in your pick functions to check if the cards were picked in the right order. You already had all the variables set up to do so!

Your HTML and CSS files was very clean and organised and follow a structured order. All the class names made sense, the HTML tags were semantic. Indentation looks good. Maybe you could add header comments to to group classes/HTML tags together. 

In the JavaScript file there were a few `console.log()` to remove. I think you could also remove some of the instructions, old comments and blank lines to make the code even cleaner. That's all to be done before being approved! 😄
```

File: code/clue.js (hunk 1 above)
Comment (ylvan75) [reply code/clue.js]: I have deleted this and other console.log that I had in the file.

File: code/clue.js (hunk 2 above)
Comment (ylvan75) [reply code/clue.js]: Removed unnecessary instructions and made sure I have my own comments, I hope it's much cleaner now.

File: code/clue.js (hunk 3 above)
Comment (ylvan75) [reply code/clue.js]: It was a mistake from from me, now fixed. Thanks for the observation.

**Review by sarahejhej — APPROVED** [review]

File: code/clue.js [hunk 17]
```diff
@@ -1,59 +1,274 @@
-// STEP 1 - CREATE OBJECTS FOR ALL THE SUSPECTS, SOMETHING LIKE THIS:
+let killerPicked = false;
+let weaponPicked = false;
+let roomPicked = false;
```
Comment (EmOlsz) [inline code/clue.js:3]: Good idea :)

File: code/clue.js [hunk 18]
```diff
@@ -1,59 +1,274 @@
...
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
 
-const rooms = []
+const randomSelector = (array) => {
```
Comment (EmOlsz) [inline code/clue.js:166]: No need for set of parentheses if you pass only one argument :

```
const randomSelector = array => { ... }
```

File: code/clue.js [hunk 19]
```diff
@@ -1,59 +1,274 @@
...
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+// Mystery object
+const mystery = {
+  killer: null,
+  weapon: null,
```
Comment (EmOlsz) [inline code/clue.js:173]: Great job again, it's the best option to initialise object with `null` values if you want to overwrite them later on :)

File: code/clue.js [hunk 20]
```diff
@@ -1,59 +1,274 @@
...
+      ).innerHTML = `Favourite Weapon: ${mystery.killer.favouriteWeapon}`;
+    }, 2000);
+    killerPicked = true;
+  }
+};
+
+// Pick weapon function
+const pickWeapon = () => {
+  if (!weaponPicked) {
+    cleanRevealMysteryMessage(); // Removes "Reveal the crime" message if any
```
Comment (EmOlsz) [inline code/clue.js:227]: 👍

File: code/clue.js [hunk 21]
```diff
@@ -1,59 +1,274 @@
...
+  }
+};
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+// Reveal function
+const revealMystery = () => {
+  if (
+    mystery.killer === null ||
+    mystery.weapon === null ||
+    mystery.room === null
```
Comment (EmOlsz) [inline code/clue.js:260]: You can simplify it :

```
  if (!mystery.killer || !mystery.weapon || !mystery.room) { ... }
```

for more information, check out https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/

File: code/index.html [hunk 22]
```diff
@@ -4,33 +4,39 @@
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <meta name="author" content="Ylva Nilsson" />
     <link rel="stylesheet" href="style.css" />
```
Comment (EmOlsz) [inline code/index.html:8]: `./` missing in path to file

File: code/index.html [hunk 23]
```diff
@@ -4,33 +4,39 @@
...
+        <p class="deck-content" id="roomName"></p>
       </article>
     </section>
 
     <h1 id="mystery"></h1>
-
-    <button>REVEAL THE CRIME</button>
+    <button id="revealCard">REVEAL THE CRIME</button>
 
     <script src="clue.js"></script>
```
Comment (EmOlsz) [inline code/index.html:41]: `./`

**Review by EmOlsz — APPROVED** [review]
Good job Ylva, congratulations on this project :) You made extra requirements and they work well!
Take a look at my comments, I hope they will help you improve even more :)

**Pull request closed without merging**